modules/Media.js

```diff
@@ -22,37 +29,75 @@ class Media extends React.Component {
     matches: true
   }
 
-  updateMatches = () =>
-    this.setState({ matches: this.mediaQueryList.matches })
+  updateMatches = () => {
+    let { query, queries } = this.props
+    if (query)
+      this.setState({
+        matches: this.mediaQueryList.reduce((accumulated, { name, mm }) => ({
+          ...accumulated,
+          [name]: mm.matches,
+        }), {}).match,
+      })
+
+    if (queries)
+      this.setState({
+        matches: this.mediaQueryList.reduce((accumulated, { name, mm }) => ({
+          ...accumulated,
+          [name]: mm.matches,
+        }), {}),
+      })
+  }
 
   componentWillMount() {
     if (typeof window !== 'object')
       return
 
-    let { query } = this.props
+    let { query, queries } = this.props
 
-    if (typeof query !== 'string')
+    if (query && typeof query !== 'string')
       query = json2mq(query)
 
-    this.mediaQueryList = window.matchMedia(query)
-    this.mediaQueryList.addListener(this.updateMatches)
+    if (query) {
+      this.mediaQueryList = [
+        {
+          name: 'match',
+          mm: window.matchMedia(query),
```

Comment:
The object returned from [`matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) is a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList), so let's call it something like `mediaQueryList` or `queryList` instead of `mm`.